File: searchingAlgo/linearSearch/linearSearch.c
#include <stdio.h>


int search(char* arr[], int lgt, char* x);

int main(void) {

	//Sample input
	char* arr[] = {"a", "b", "c", "d"};
	char* x = "b";


	int lgt = sizeof(arr)/sizeof(arr[0]);
	printf("%i\n", search(arr, lgt, x));
}


int search(char* arr[], int lgt, char* x) {
	for(int i = 0; i < lgt; i++){
		if (arr[i] == x){
			return i;
		}
	}
	return -1;
}

File: searchingAlgo/linearSearch/linearSearch.js
function search(arr, x){
	for(i = 0, lgt = arr.length; i < lgt; i++) {
		if(arr[i] == x) return i;
	}
	return -1
}

// Sample input
arr = ["a", "b", "c", "d"];
x = "b";

console.log(search(arr, x));
